### PR TITLE
open3d_show_and_abort_on_warning(Core)

### PR DESCRIPTION
**Pull request by yxlao**
This change is [Reviewable](https://reviewable.io/reviews/intel-isl/open3d/1959)

### CODE REVIEW

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5694490&bid=18636426)

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5694490&bid=18636426)

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5694490&bid=18636426)

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5694490&bid=18636426)

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5694490&bid=18636426)

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+    # nvcc requires an option after -Werror
+    # https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#generic-tool-options-Werror
+    target_compile_options(${trgt} PRIVATE
+        $<$<COMPILE_LANGUAGE:C>:$<$<CXX_COMPILER_ID:MSVC>:/W3>>
+        $<$<COMPILE_LANGUAGE:C>:$<$<CXX_COMPILER_ID:Clang>:-Wall -Werror>>
+        $<$<COMPILE_LANGUAGE:C>:$<$<CXX_COMPILER_ID:GNU>:-Wall -Werror>>
+        $<$<COMPILE_LANGUAGE:CXX>:$<$<CXX_COMPILER_ID:MSVC>:/W3>>
+        $<$<COMPILE_LANGUAGE:CXX>:$<$<CXX_COMPILER_ID:Clang>:-Wall -Werror>>
+        $<$<COMPILE_LANGUAGE:CXX>:$<$<CXX_COMPILER_ID:GNU>:-Wall -Werror>>
+        $<$<COMPILE_LANGUAGE:CUDA>:-Wall -Werror all-warnings>
```
Comment (pvarvak) [inline src/CMakeLists.txt:11]: shall we remove -Wall for CUDA?

Comment (yxlao) [reply src/CMakeLists.txt]: yes, done.

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5694490&bid=18636426)

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5694490&bid=18636426)

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5694490&bid=18636426)

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5694490&bid=18636426)

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5694490&bid=18636426)

**Pull request merged**